HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+
+In future the network could adopt a system contract function that supports maxAutoAssociation changes
+
+### System Contract Functions
+
+HTS system contracts will need to support the new airdrop related transactions to ensure the feature is exposed via smart contract functions. Additionally, redirect functions will be provided to ensure optimal UX in which EOAs can initiate transactions through a DApp that affect their account.
+
+| selector | function | IHRC |
+| --- | --- | --- |
+| 0x… | <htsSystemContractAddress>.airdropTokens(TokenTransferList[] memory tokenTransfers) |  |
+| 0x… | <htsTokenAddress>.cancelAirdropFT(address receiverAddress) | Y |
+| 0x… | <htsTokenAddress>.cancelAirdropNFT(address receiverAddress, int64 serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.cancelAirdrops(TokenPendingAirdrop[] memory pendingAirdrops) |  |
+| 0x… | <htsTokenAddress>.claimAirdropFT(address senderAddress) | Y |
+| 0x… | <htsTokenAddress>.claimAirdropNFT(address senderAddress, int64 serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.claimAirdrops(TokenPendingAirdrop[] memory pendingAirdrops) |  |
+| 0x… | <htsTokenAddress>.rejectTokenFT() | Y |
+| 0x… | <htsTokenAddress>.rejectTokensNFT(int64[] serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.rejectTokens(address[] memory ftAddresses, NftId[] memory nftIds) |  |
+| 0x… | <hasSystemContractAddress>.setAutomaticAssociations(boolean enableAutoAssociations) | Y |
+
+To increase API reuse and for consistency the input objects will closely match the HAPI protobuf objects.
+
+### SDK
+
+The SDKs must support the new HAPI transactions
+
+- `TokenAirdrop` transaction - exposes `TokenAirdropTransactionBody` proto
+- `TokenCancel` transaction - exposes `TokenCancelAirdropTransactionBody` proto
+- `TokenClaim` transaction - exposes `TokenClaimAirdropTransactionBody` proto
+- `TokenReject` transaction - exposes `TokenRejectTransactionBody` proto
+- `Token?Flow` that supports rejection and dissociation to make it easier for devs
+
+The SDKs must update the `TransactionRecord` to return the following information
+- `new_pending_airdrops`
+
+### MAPI (Mirror Node API)
+
+- New `/api/v1/accounts/{senderIdOrEvmAddress}/airdrops/outstanding` - List of outstanding airdrops sent by `senderIdOrEvmAddress` which have not been claimed by recipients
+    
+    ```json
+       {
+    			"airdrops": [
+            {
+                "amount": 333,
+                "receiver_id": "0.0.999",
+                "sender_id": "0.0.222",
+    						"serial_number": null,
+                "token_id": "0.0.111"
+            },
+            {
+                "amount": 555,
+                "receiver_id": "0.0.999",
+                "sender_id": "0.0.222",
+    						"serial_number": null,
+                "token_id": "0.0.444"
+            },
+            {
+                "amount": null,
+                "receiver_id": "0.0.999",
+                "sender_id": "0.0.777",
```

Comment:
Shouldn't the response have contain entries with the same `sender_id` (i.e. given the two previous entries I would expect 0.0.222 here as well)?
And if the response should contain entries with the same `sender_id` isn't that redundant and can we instead omit it from the response body?